### PR TITLE
Pavel refactor user service & user controller

### PR DESCRIPTION
**Pull request by 2HEADSs**
Refactor userController and userService to work with 3 different roles
Remove unecessary logs
Add TODO's for future development

### CODE REVIEW

File: src/controllers/userController.js [hunk 1]
```diff
@@ -12,17 +12,32 @@ userController.post('/registerUser', async (req, res) => {
...
+        */
+        if (req.body.role == 'admin') {
+            throw new Error('You do not have admin rights!');
+        }
+
         const userData = {
             //TODO: TEST if this check is needed
             email: req.body.email,
-            organizator: req.body.organizator ? req.body.organizator : '',
+            organizatorName: req.body.organizatorName
```
Comment (Getsov) [inline src/controllers/userController.js:30]: check by design, it must be required, don't set it to empty string if it is an organizator and it mustn't apply if regular/admin. It applies for all below, check the design

File: src/controllers/userController.js [hunk 2]
```diff
@@ -100,7 +115,23 @@ userController.put('/editUserPassword/:id', async (req, res) => {
...
 module.exports = {
     userController,
 };
+
+
+
+/*
+Ready user for register
+    "email": "pavel@abv.bg",
+    "organizatorName": "Suzuki Burgas",
```
Comment (Getsov) [inline src/controllers/userController.js:129]: there shouldn't be an organizatorName for regular user

File: src/services/userService.js [hunk 3]
```diff
@@ -71,15 +72,24 @@ async function updateUserInfo(userId, requestBody, isAdmin) {
...
     }
 
+    /*
+    // if (requestBody.role == 'organizer' || requestBody.role == 'regular') {
+    //     existingUser.role = requestBody.role;
+    // }
+     */
+
     if (isAdmin) {
+        'isDleted' in requestBody
```
Comment (Getsov) [inline src/services/userService.js:82]: isDeleted

File: src/controllers/userController.js (hunk 1 above)
Comment (2HEADSs) [reply src/controllers/userController.js]: fixed

File: src/services/userService.js (hunk 3 above)
Comment (2HEADSs) [reply src/services/userService.js]: fixed

File: src/controllers/userController.js (hunk 2 above)
Comment (2HEADSs) [reply src/controllers/userController.js]: fixed

**Review by Getsov — APPROVED** [review]

**Pull request merged**